app/views/new_gestionnaire/dossiers/_pieces_jointes.html.haml

```diff
@@ -12,9 +12,22 @@
     - dossier.procedure.types_de_piece_justificative.each do |type_de_piece_justificative|
       %tr
         %th= "#{type_de_piece_justificative.libelle} :"
-        %td
-          - pj = dossier.retrieve_last_piece_justificative_by_type(type_de_piece_justificative.id)
-          - if pj.present?
-            = link_to "Télécharger", pj.content_url, class: "link", target: :blank
-          - else
-            Pièce non fournie
+        - pjs = dossier.retrieve_all_piece_justificative_by_type(type_de_piece_justificative.id).to_ary.dup
```

Comment:
Juste le dup ça coute pas cher ? par rapport à itérer avec un index sur l'association de base ?

Reply:
nop, ça coute pas cher